system-test/spanner.ts

```diff
@@ -1558,17 +1572,17 @@ describe('Spanner', () => {
                   child: 'bool',
                 },
               },
-            };
+            } as {}) as ExecuteSqlRequest;
```

Comment:
Any reason we have to cast it like this? I would think `const query: ExecuteSqlRequest = {};` would work

Reply:
Follow up - assuming that does work, we should change all the fixtures that use this form of casting.

Reply:
tried with that already but gives below Error:
`Type '{ v: { type: string; child: string; }; }' is not assignable to type '{ [param: string]: string; }'.
 Property 'v' is incompatible with index signature.
 Type '{ type: string; child: string; }' is not assignable to type 'string'`

also tried with `as ExecuteSqlRequest` only but that is also giving error.

Reply:
I see, it looks like the [typing](https://github.com/googleapis/nodejs-spanner/blob/master/src/transaction.ts#L60) is incorrect. We should probably fix it to allow for it to be a [`Type` object](https://github.com/googleapis/nodejs-spanner/blob/master/src/codec.ts#L397)